packages/typespec-python/src/emitter.ts

```diff
@@ -373,7 +403,11 @@ function emitResponse(context: SdkContext, response: HttpOperationResponse): Rec
     if (body) {
         if (body.kind === "Model") {
             if (body && body.decorators.find((d) => d.decorator.name === "$pagedResult")) {
-                type = getType(context, Array.from(body.properties.values())[0]);
+                if (modelsMode !== "msrest") {
+                    type = getType(context, Array.from(body.properties.values())[0]);
```

Comment:
There is difference about `extract_data` in paging operation between [dpg models](https://github.com/Azure/azure-sdk-for-python/blob/5fdb5ac43b5ac81f336aed27216695a3fb4db37d/sdk/sphere/azure-mgmt-sphere/azure/mgmt/sphere/_operations/_operations.py#L1628-L1633) and [msrest models](https://github.com/Azure/autorest.python/blob/fa2742f94e1dd724e64ff5832f690fdb7bf8a7d1/packages/typespec-python/test/generated/azure-mgmt-spheremsrest/azure/mgmt/spheremsrest/_operations/_operations.py#L1554-L1559). The code here is to handle it.